src/authentication/oauth2Auth.ts

```diff
@@ -245,14 +245,22 @@ export class Oauth2Auth extends AlfrescoApiClient {
                     reject('Validation JWT error' + error);
                 });
             } else {
-                if (this.config.oauth2.silentLogin) {
+                if (this.config.oauth2.silentLogin && !this.isPublicUrl()) {
                     this.implicitLogin();
                 }
+                resolve();
             }
         });
 
     }
 
+    isPublicUrl(): boolean {
+        const publicUrls = this.config.oauth2.publicUrls || [];
+
+        return publicUrls.length &&
+            publicUrls.some((url: string) => window.location.href.startsWith(url));
```

Comment:
@eromano @DenysVuika are we sure we want `startsWith` and not `contains` ?

Reply:
maybe something based on regex could it be better but with less performance, do we want to start trying it like that?

Reply:
i would go with `includes`, covers partial and full string. we can opt for regex if we find that can help us cover more scenarios

Reply:
should we go directly for regex?

Reply:
can we use `minimatch` for this? I believe if we have a public url as **`**/preview/s/*`** and href  `some/long/url/preview/s/1234` this will match and it would work with ng router paths like `preview/s/:nodeId`